packages/upload/src/vaadin-upload.js

```diff
@@ -446,6 +453,36 @@ class Upload extends ElementMixin(ThemableMixin(PolymerElement)) {
     this.addEventListener('upload-start', this._onUploadStart.bind(this));
     this.addEventListener('upload-success', this._onUploadSuccess.bind(this));
     this.addEventListener('upload-error', this._onUploadError.bind(this));
+
+    this.addController(
+      new SlotController(
+        this,
+        'add-button',
+        () => document.createElement('vaadin-button'),
+        (_, button) => {
+          button.addEventListener('touchend', (e) => {
+            this._onAddFilesTouchEnd(e);
+          });
+          button.addEventListener('click', (e) => {
+            this._onAddFilesClick(e);
+          });
+          this._addButton = button;
```

Comment:
`_addButton` should probably also be declared as a private Polymer property.

Reply:
Thanks, done.